lib/datadog/profiling/component.rb

```diff
@@ -110,6 +112,22 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
           )
       end
 
+      private_class_method def self.build_crashtracker(settings, transport)
+        return unless settings.profiling.advanced.experimental_crash_tracking_enabled
+
+        unless transport.respond_to?(:exporter_configuration)
```

Comment:
Actually, I'm not quite sure:

> * A refactoring of the default transport could lead to a situation where a warning starts being emitted about usage of custom transport when that is not the case.

Yes, but our tests would catch that (I actually went in and just tried it). Specifically

> 1) Datadog::Profiling::Component.build_profiler_component with :enabled true when crash tracking is enabled initializes the crash tracker
> 2) Datadog::Profiling::Component.build_profiler_component with :enabled true when crash tracking is enabled initializes the profiler instance with the crash tracker
> 3) Datadog::Profiling::HttpTransport#exporter_configuration returns the current exporter configuration

both fail because on top of the warning, the crash tracker doesn't get enabled. So I think this covers us in terms of "oops we changed the API and forgot about this detail".

> * A custom transport could be transparently used with no warnings even though it may not actually be fully compatible with crash tracking.

Note that this check happens only when the crash tracker is enabled, so if you have a custom transport that doesn't expose `exporter_configuration` you'll not get a warning by default.

And if you do see the warning, you can either expose an `exporter_configuration` on the custom transport OR disable the crash tracker to make the issue go away.